common/persistence/cassandra/queue_store_v2.go

```diff
@@ -0,0 +1,356 @@
+		queueMessage := convertQueueV2Message(message)
+		result = append(result, *queueMessage)
+		message = make(map[string]interface{})
+	}
+
+	var nextPageToken []byte
+	if len(iter.PageState()) > 0 {
+		nextPageToken = iter.PageState()
+	}
+
+	if err := iter.Close(); err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("ReadMessages operation failed. Error: %v", err))
+	}
+
+	pageToken.PageToken = nextPageToken
+	pageTokenByte, err := gobSerialize(pageToken)
+	if err != nil {
+		return nil, err
+	}
+	return &persistence.InternalReadMessagesResponse{Messages: result,
+		NextPageToken: pageTokenByte,
+	}, nil
+}
+
+func (q *QueueStoreV2) RangeDeleteMessages(ctx context.Context, request persistence.InternalRangeDeleteMessagesRequest) (*persistence.InternalRangeDeleteMessagesResponse, error) {
+	ackLevel, version, err := q.getCurrentAckLevelAndVersion(ctx, request.QueueType, request.QueueName)
+	if err != nil {
+		return nil, err
+	}
+
+	query := q.session.Query(templateDeleteMessagesQueryV2, request.QueueType, request.QueueName, ackLevel, request.InclusiveMaxMessageMetadata.ID).WithContext(ctx)
+	if err := query.Exec(); err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("RangeDeleteMessagesFromDLQ operation failed. Error %v, Type: %v, Name: %v", err, request.QueueType, request.QueueName))
+	}
+
+	return &persistence.InternalRangeDeleteMessagesResponse{}, q.updateAckLevel(ctx, request, version)
+}
+
+func (q *QueueStoreV2) ListQueues(ctx context.Context, request persistence.InternalListQueuesRequest) (*persistence.InternalListQueuesResponse, error) {
+	query := q.session.Query(templateGetQueuesListQueryV2, request.QueueType).WithContext(ctx)
+	iter := query.PageSize(request.PageSize).PageState(request.NextPageToken).Iter()
+
+	response := persistence.InternalListQueuesResponse{}
+	queueName := make(map[string]interface{})
+	for iter.MapScan(queueName) {
+		response.QueueNames = append(response.QueueNames, queueName["queue_name"].(string))
+		queueName = make(map[string]interface{})
+	}
+
+	if len(iter.PageState()) > 0 {
+		response.NextPageToken = iter.PageState()
+	}
+
+	if err := iter.Close(); err != nil {
+		return nil, serviceerror.NewUnavailable(fmt.Sprintf("ListQueues operation failed. Error: %v, Type: %v", err, request.QueueType))
+	}
+
+	return &response, nil
+}
+
+func (q *QueueStoreV2) Close() {
+	if q.session != nil {
+		q.session.Close()
+	}
+}
+
+func (q *QueueStoreV2) tryEnqueue(
+	ctx context.Context,
+	request persistence.InternalEnqueueMessageRequest,
+	messageID int64,
+	partition int,
+) (int64, error) {
+	query := q.session.Query(templateEnqueueMessageQuery, request.QueueType, request.QueueName, partition, messageID, request.Blob.Data, request.Blob.EncodingType.String()).WithContext(ctx)
+	previous := make(map[string]interface{})
+	applied, err := query.MapScanCAS(previous)
+	if err != nil {
+		return persistence.EmptyQueueMessageID, gocql.ConvertError("tryEnqueue", err)
```

Comment:
shall we use ConvertError in other places of this file?

Reply:
very good question 🤔 thinking out loud here. In the implementation the default case has sth like this: 
```
serviceerror.NewUnavailable(fmt.Sprintf("operation %v encountered %v", operation, err.Error()))
```
Which means if the errors are not known, it will return a `NewUnavailable` along with the name of the higher level operation and the lower level error. But For some of these errors, we might want to provide more details? like the type of the queue and the name of the queue along with the operation? That helps with identifying the specific queue that is encountering an issue. 

But at the same time, `ConvertError` helps to produce the proper error in cases other than `default`. 

so maybe a middle ground is to include those information in the `operation` we pass to `ConvertError`? like `tryEnqueue(type=QueueType, name=QueueName)`.